test/ctia/bundle/core_asset_ref_entities_test.clj

```diff
@@ -0,0 +1,197 @@
+                                bulk [:asset_mappings
+                                      :asset_properties])))
+                          ;; Doesn't make sence to continue from here; once we
+                          ;; asserted that asset-mapping and asset-properties
+                          ;; won't be explicitly created (until we create Assets
+                          ;; with non-transient IDs), we achieved the goal of
+                          ;; this test.
+                          (throw (Exception. "stopped intentionally")))]
+            (is
+             (thrown-with-msg?
+              Exception #"stopped intentionally"
+              (bulk/create-bulk
+               bulk
+               tempids
+               login
+               {}
+               services))))))))))
+
+(use-fixtures :once
+  (join-fixtures
+   [whoami-helpers/fixture-server]))
+
+(deftest asset-refs-test
+  (th/fixture-ctia-with-app
+   (fn [app]
+     (th/set-capabilities! app "foouser" ["foogroup"] "user" auth/all-capabilities)
+     (whoami-helpers/set-whoami-response app
+                                         "45c1f5e3f05d0"
+                                         "foouser"
+                                         "foogroup"
+                                         "user")
+     (let [services (app/service-graph app)
+
+           {{:keys [get-store]} :StoreService} services
+
+           login              (map->Identity {:login  "foouser"
+                                              :groups ["foogroup"]})
+           bundle-import-data (bundle/prepare-import bundle-ents nil login services)
+           bulk               (bundle/prepare-bulk bundle-import-data)
+           tempids            (->> bundle-import-data
+                                   (map (fn [[_ entities-import-data]]
+                                          (bundle/entities-import-data->tempids
+                                           entities-import-data)))
+                                   (apply merge {}))
+           {:keys [tempids]}  (bulk/create-bulk
+                               bulk
+                               tempids
+                               login
+                               {:refresh true}
+                               services)
+           get-records        (fn [store]
+                                (-> (get-store store)
+                                    (store/list-records
+                                     {:query "*"}
+                                     {:login  "foouser"
+                                      :groups ["foogroup"]} {})
+                                    :data))
+           expected-asset-ref (get tempids (-> bundle-ents :assets first :id))]
+       (testing "every AssetMapping and AssetProperties have proper
+                 non-transient :asset-ref"
+        (are [store] (every?
+                      (partial = expected-asset-ref)
+                      (->> store get-records (map :asset_ref)))
```

Comment:
You can `assert seq` the result of `get-records`, otherwise `every?` can return `true` for an empty coll